views/js/pciCreator/dev/audioRecordingInteraction/runtime/js/providers/webAudio.js

```diff
@@ -0,0 +1,185 @@
+        // Leaving it as it is for now as the primary use case for uncompressed recording is offline testing.
+        var recorderWorkerPath = 'audioRecordingInteraction/runtime/js/workers/WebAudioRecorderWav.js',
+            recorderWorker;
+
+        var audioContext,
+            audioNodes = {};
+
+        var numChannels = (config.isStereo) ? 2 : 1,
+            buffer = [];
+
+        /**
+         * Load the worker and configure it
+         */
+        function initWorker() {
+            recorderWorker = new Worker(assetManager.resolve(recorderWorkerPath));
+
+            sendToWorker('init', {
+                config: {
+                    numChannels: numChannels,
+                    sampleRate: audioContext.sampleRate
+                },
+                options: {
+                    timeLimit: 0,           // time limit is handled by the provider wrapper
+                    progressInterval: 1000, // encoding progress report interval (millisec)
+                    wav: {
+                        mimeType: "audio/wav"
+                    }
+                }
+            });
+        }
+
+        /**
+         * Execute a command on the worker. See workers/WebAudioRecorderWav.js for a list of commands and their parameters
+         * @param {String} command
+         * @param {Object} payload - command parameters, depends on the actual command
+         */
+        function sendToWorker(command, payload) {
+            recorderWorker.postMessage(_.merge({ command: command }, payload));
+        }
+
+        /**
+         * The provider
+         */
+        webAudioProvider = {
+            /**
+             * Expose the Audio context so consumers can use it instead of creating a new one
+             * @returns {AudioContext}
+             */
+            getAudioContext: function getAudioContext() {
+                return audioContext;
+            },
+
+            /**
+             * Create the base audio nodes and add listeners for worker commands
+             * @param {MediaStream} stream
+             */
+            init: function init(stream) {
+                var self = this;
+
+                audioContext = new (window.AudioContext || window.webkitAudioContext)();
+                audioNodes.source = audioContext.createMediaStreamSource(stream);
+                audioNodes.inputGain = audioContext.createGain();
+
+                audioNodes.source.connect(audioNodes.inputGain);
+
+                initWorker();
+
+                recorderWorker.onmessage = function(e) {
+                    var data = e.data;
+                    var blob;
+                    switch (data.command) {
+                        case 'complete': {
+                            blob = data.blob;
+                            self.trigger('blobavailable', [blob]);
+                            break;
+                        }
+                        case 'error': {
+                            window.console.error(data.message);
```

Comment:
For now there is no standard way of managing errors in PCIs, mainly due to the fact that PCIs are not localised yet, making it difficult to provide relevant information to the test taker.

I will remove this listener as the only case where an error is triggered in the worker is by calling a setOption() during recording, which is a method we don't use for now. If ever we start using that method, we should put it back and implement an error feedback strategy at this stage.

Regarding `onerror` I'm not sure we should override the global handler it in a PCI, if each PCI starts doing that we will end up with conflicts. Also, the worker not being loaded is not an error the user can act upon. He could retry in case of an network error but surprisingly, no error is triggered in that case (https://developer.mozilla.org/en-US/docs/Web/API/Worker/Worker)